core/primitives/impl/scale_codec_impl.cpp

```diff
@@ -155,6 +157,30 @@ namespace kagome::scale {
       return collection::decodeCollection<uint8_t>(stream);
     }
   };
+
+  /// decodes primitives::parachain::Chain
+  template <>
+  struct TypeDecoder<primitives::parachain::Chain> {
+    outcome::result<primitives::parachain::Chain> decode(
+        common::ByteStream &stream) {
+      return variant::decodeVariant<primitives::parachain::Relay,
+                                    primitives::parachain::Parachain>(stream);
+    }
+  };
+
+  /// decodes object of empty class
+  template <class T>
+  struct EmptyDecoder {
+    outcome::result<T> decode(common::ByteStream &) {
+      return T{};
+    }
+  };
+
+  /// decodes fake type primitives::parachain::Relay, which has no content
+  template <>
+  struct TypeDecoder<primitives::parachain::Relay>
+      : public EmptyDecoder<primitives::parachain::Relay> {};
```

Comment:
where is implementation?

Reply:
public EmptyEncoder<primitives::parachain::Relay>